geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java

```diff
@@ -578,6 +625,91 @@ private int getMaxElementsToReturn(AbstractSortedSetRangeOptions<?> rangeOptions
     return result;
   }
 
+  private RedisSortedSet getIntersection(List<RedisSortedSet> sets, ZAggregator aggregator) {
+    RedisSortedSet retVal = new RedisSortedSet(Collections.emptyList());
+
+    for (RedisSortedSet set : sets) {
+      for (OrderedSetEntry entry : set.members.values()) {
+        Double newScore;
+        if (aggregator.equals(ZAggregator.SUM)) {
+          newScore = recursivelySumScoresForMember(sets, entry.member, 0D);
+        } else if (aggregator.equals(ZAggregator.MAX)) {
+          newScore = recursivelyGetMaxScoreForMember(sets, entry.member, Double.MIN_VALUE);
+        } else {
+          newScore = recursivelyGetMinScoreForMember(sets, entry.member, Double.MAX_VALUE);
+        }
+
+        if (newScore != null) {
+          if (newScore.isNaN()) {
+            throw new ArithmeticException(ERROR_OPERATION_PRODUCED_NAN);
+          }
+          retVal.memberAdd(entry.getMember(), Coder.doubleToBytes(newScore));
```

Comment:
It seems like if you passed retVal in to recursively*ScoreForMember then it could do the retVal.memberAdd if needed. That would also allow us to get rid of this use of "Double". If you get rid of the recursion then these methods could just take a member, input sets, and a set to fill.